src/server/req_runjob.c

```diff
@@ -452,27 +452,26 @@ req_runjob(struct batch_request *preq)
 	DBPRT(("req_runjob: received command to run job on destin %s\n",
 		preq->rq_ind.rq_run.rq_destin))
 
-	switch (process_hooks(preq, hook_msg, sizeof(hook_msg),
-			pbs_python_set_interrupt)) {
-
-		case 0:      /* explicit reject */
-			reply_text(preq, PBSE_HOOKERROR, hook_msg);
-			return;
-		case 1:    /* explicit accept */
-		case 2:    /* no hook script executed - go ahead and accept events*/
-			break;
-		default:
-			log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_HOOK,
-				LOG_INFO, "", "runjob event: accept req by default");
-	}
-
 	/* OK - go back over the run job request, assign the vhosts */
 	/* and finally run the job by calling req_runjob2()	    */
 
 	if (jt == IS_ARRAY_NO) {
 
 		/* just a regular job, pass it on down the line and be done */
 
+		switch (process_hooks(preq, hook_msg, sizeof(hook_msg),
+				pbs_python_set_interrupt)) {
+
+			case 0:      /* explicit reject */
+				reply_text(preq, PBSE_HOOKERROR, hook_msg);
+				return;
+			case 1:    /* explicit accept */
+			case 2:    /* no hook script executed - go ahead and accept events*/
+				break;
+			default:
+				log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_HOOK,
+					LOG_INFO, "", "runjob event: accept req by default");
+		}
```

Comment:
Please put this block of code in a function, checking for the return value of '0' to issue reply_text(PBSE_HOOKERROR). as I see this same code gets called in 2 other places.